### ISSUE TITLE
4.8.0 install hangs on Node 6/NPM 3 on Windows

### ISSUE THREAD

**Issue by Robin-Law** [webpack/webpack#7228]
# Bug report

**What is the current behavior?**
With the 4.8.0 release of webpack, we are seeing builds timeout on our CI server (VSTS) during `npm install -g webpack`. This also happens on a local machine running the same version of node/npm. 4.7.0 and 4.6.0 install without issue.

**If the current behavior is a bug, please provide the steps to reproduce.**

Steps to reproduce:
1. Use node 6.9.1 and npm 3.10.8 on Windows.
2. `npm install -g webpack`

**What is the expected behavior?**
Running `npm install -g webpack` under node 6 resolves and installs the package normally in a reasonable amount of time (<1 minute).

**Other relevant information:**
webpack version:  4.8.0
Node.js version: 6.9.1 (npm 3.10.8)
Operating System: Windows
Additional tools:

**Comment by Robin-Law** [comment]
The logs around the time of the hang:

```
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\webpack-be7c0b4a
npm verb tar unpack C:\Users\robin\AppData\Roaming\npm-cache\@webassemblyjs\wasm-edit\1.3.0\package.tgz
npm verb tar unpacking to C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-edit-f27b1409
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-edit-f27b1409
npm verb tar unpack C:\Users\robin\AppData\Roaming\npm-cache\webpack\4.8.0\package.tgz
npm verb tar unpacking to C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\webpack-be7c0b4a
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\webpack-be7c0b4a
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\helper-wasm-bytecode-6191f139\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\helper-fsm-312ca601\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\helper-code-frame-d3af758c\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\helper-buffer-52c22181\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-opt-80c25b2a\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-gen-7240019a\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\leb-f2d564d0\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wast-printer-6aeff993\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\floating-point-hex-parser-e085394c\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\helper-wasm-section-a32306d7\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-edit-f27b1409\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\leb128-86560c47\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\debug-f3c6d0ac\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\validation-e436635f\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\debug-9d7b0a52\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\debug-06359c1d\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\long-0c83c8d0\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\ast-9d7a5a3d\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\@webassemblyjs\wasm-parser-2ed91fbe\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\webassemblyjs-de5669ca\node_modules
npm verb gentlyRm don't care about contents; nuking C:\Users\robin\AppData\Roaming\npm\node_modules\.staging\webpack-be7c0b4a\node_modules
```

**Comment by tstewart15** [comment]
I was about to file a bug report as well. Only difference in my case is that I'm using node v7.x.x and npm v4.2.0.

**Comment by sokra** [comment]
@xtuc Is there anything special with your packages?

**Comment by xtuc** [comment]
We have some cyclic depency issues,  I doubt it causes this.

**Comment by xtuc** [comment]
Webpack also uses a Windows based CI, right?

**Comment by sokra** [comment]
yep, but we use yarn to install

**Comment by sokra** [comment]
I tried with node.js 6.14.1 and npm 3.10.10 on Windows 10 and it worked fine.

I also tried with node.js 6.9.1 and npm 3.10.8 and it also worked.

**Comment by evilebottnawi** [comment]
Looks like something invalid with node installation and/or npm, or other os environment problems
@xtuc maybe more information about os will help solve problem/detect. 

1. What is windows and version (example as here https://www.lifewire.com/windows-version-numbers-2625171)? 

2. Also try to run `npm doctor`.

3. Try to install https://github.com/coreybutler/nvm-windows and switch to node/npm version

**Comment by sokra** [comment]
Using webpack on node.js 6 is not recommended anyway, as it's super slow... Upgrade your node.js installation.

**Comment by tstewart15** [comment]
My issue has been resolved, perhaps related to 4.8.1 release; webpack v4.8.1 was successfully updated to (from v4.7.0) using `npm up`, npm v4.2.0, node v7.9.0

**Comment by evilebottnawi** [comment]
Looks like we can close issue with recommendation to upgrade `4.8.1` node. ~~Maybe something broken in `node` and/or npm (or together)~~ https://github.com/npm/npm/issues/13722. We can't do anything in webpack with this, sorry. Thanks everyone! Feel free to feedback.
